sqlglot/parser.py

```diff
@@ -1287,23 +1288,30 @@ def _parse_lateral(self):
             return None
 
         if not this:
-            this = self._parse_function()
-
-        table_alias = self._parse_id_var(any_token=False)
+            this = self._parse_function() or self._parse_id_var(False)
+            while self._match(TokenType.DOT):
+                this = exp.Dot(this=this, expression=self._parse_function() or self._parse_id_var())
 
         columns = None
-        if self._match(TokenType.ALIAS):
-            columns = self._parse_csv(self._parse_id_var)
-        elif self._match(TokenType.L_PAREN):
-            columns = self._parse_csv(self._parse_id_var)
-            self._match_r_paren()
+        table_alias = None
+        if view or self.TREAT_LAT_FUNC_AS_LAT_VIEW:
+            table_alias = self._parse_id_var(any_token=False)
+            if self._match(TokenType.ALIAS):
+                columns = self._parse_csv(self._parse_id_var)
+        else:
+            self._match(TokenType.ALIAS)
+            table_alias = self._parse_id_var(any_token=False)
+
+            if self._match(TokenType.L_PAREN):
+                columns = self._parse_csv(self._parse_id_var)
+                self._match_r_paren()
 
         expression = self.expression(
             exp.Lateral,
             this=this,
             view=view,
             outer=outer,
-            alias=self.expression(exp.TableAlias, this=table_alias, columns=columns),
+            alias=exp.TableAlias(this=table_alias, columns=columns),
```

Comment:
Why was the `self.expression` call removed here? We try to use this as much as possible because it includes some logic for comment attaching etc.

Reply:
Reverted that change